src/onnx/parse_slice.cpp

```diff
@@ -69,8 +111,19 @@ struct parse_slice : op_parser<parse_slice>
         if(args.size() >= 3)
         {
             migraphx::argument end_arg = args.at(2)->eval();
-            check_arg_empty(end_arg, "PARSE_SLICE: cannot handle variable ends for slice");
-            end_arg.visit([&](auto s) { op.ends.assign(s.begin(), s.end()); });
+            if(end_arg.empty())
+            {
+                op_args.insert(op_args.begin(), args.at(2));
+                if(not steps.empty())
+                {
+                    MIGRAPHX_THROW(
+                        "PARSE_SLICE: steps and variable starts and ends is not supported");
+                }
+            }
+            else
+            {
+                end_arg.visit([&](auto s) { op.ends.assign(s.begin(), s.end()); });
+            }
```

Comment:
When you make the return a struct you can add a method to do this pattern:

```cpp
struct slice_desc
{
    std::vector<instruction_ref> op_args;
    std::vector<int64_t> steps;
    std::vector<int64_t> raxes;

    std::vector<int64_t> insert(instruction_ref arg)
    {
        std::vector<int64_t> result;
        migraphx::argument arg_value = arg->eval();
        if(arg_value.empty())
        {
            op_args.insert(op_args.begin(), arg);
        }
        else
        {
            arg_value.visit([&](auto s) { result.assign(s.begin(), s.end()); });
        }
        return result;
    }
};
```

Then you can just write the above as `op.ends = sd.insert(args.at(2))`. This should get rid of the extra branching so it will pass tidy.

Also, I think you could check for the `steps` afterwards:

```cpp
if (not steps.empty())
{
    if (op.starts.empty() or op.ends.empty())
        MIGRAPHX_THROW("PARSE_SLICE: steps and variable starts and ends is not supported");
    if (op.axes.empty())
        MIGRAPHX_THROW("PARSE_SLICE: steps and variable axes is not supported");
}
```